Source/Core/Repositories/OrganizationRepository.cs

```diff
@@ -223,6 +218,11 @@ public class OrganizationRepository : RepositoryBase<Organization>, IOrganizatio
                 await Cache.SetAsync(GetUsageSavedCacheKey(organizationId), DateTime.UtcNow, TimeSpan.FromDays(32)).AnyContext();
             }
 
+            if (justWentOverMonthly)
+                await PublishMessageAsync(new PlanOverage { OrganizationId = org.Id }).AnyContext();
+            else if (justWentOverHourly)
+                await PublishMessageAsync(new PlanOverage { OrganizationId = org.Id, IsHourly = true }).AnyContext();
+
```

Comment:
I moved this down to after the organization is saved. This ensures that when the message is published and the organization is retrieved the stats are up-to-date. So we could give detailed and up-to-date metrics when you go over your limit.